op-e2e/e2eutils/geth/wait.go

```diff
@@ -100,23 +99,20 @@ func WaitForBlock(number *big.Int, client *ethclient.Client, timeout time.Durati
 			}
 		case err := <-headSub.Err():
 			return nil, fmt.Errorf("error in head subscription: %w", err)
-		case <-timeoutCh:
```

Comment:
We can remove the `timeoutCh`, but we might still need to wait on `<-ctx.Done()` here and in that case return `nil, ctx.Err()`. IIRC the context passed to `client.SubscribeNewHead(ctx, headChan)` is only used for _establishing_ the subscription, but is not used any more when the sub is running.